flink-connectors/flink-connector-kafka-base/src/test/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaConsumerBaseTest.java

```diff
@@ -265,10 +274,8 @@ public void testSnapshotStateWithCommitOnCheckpointsEnabled() throws Exception {
 
 		// --------------------------------------------------------------------
 
-		final OneShotLatch runLatch = new OneShotLatch();
-		final OneShotLatch stopLatch = new OneShotLatch();
-		final AbstractFetcher<String, ?> fetcher = getRunnableMockFetcher(runLatch, stopLatch);
-		when(fetcher.snapshotCurrentState()).thenReturn(state1, state2, state3);
+		final MockFetcher<String> fetcher = spy(new MockFetcher<>());
+		doReturn(state1).doReturn(state2).doReturn(state3).when(fetcher).snapshotCurrentState();
```

Comment:
Maybe we could go one small step further?

```
private static class MockFetcher<T> ... {
  private final ArrayDeque<HashMap<KafkaTopicPartition, Long>> stateSnapshotsToReturn = new ArrayDeque<>();
  
  public MockFetcher(HashMap<KafkaTopicPartition, Long>.. stateSnapshotsToReturn) {
    this.stateSnapshotsToReturn.addAll(Arrays.asList(stateSnapshotsToReturn));
  }

  @Override
  public HashMap<KafkaTopicPartition, Long> snapshotCurrentState() {
    checkState(!stateSnapshotsToReturn.isEmpty());
    return stateSnapshotsToReturn.poll();
  }
}
```

Reply:
Yes, if we really want to get rid of Mockito for good :) 

Though `checkState(!stateSnapshotsToReturn.isEmpty())` is a bit problematic, because the unit tests rely on the last return stub to re-occur for all remaining calls.

I'll change this while merging.

Reply:
Ops, haven't noticed that :)